default.nix

```diff
@@ -372,6 +373,47 @@ let
       });
     };
 
+    marlowe-symbolic-lambda = 
+	    let
+        # NOTE: Unfortunately we cannot build a static executable if the package uses template haskell.
+        # In order to get around this we make sure that the TH code is all in libraries and create
+        # a single haskell file that does nothing other than import the main function from the library.
+        # We can then statically compile this as it does not use TH.
```

Comment:
nope, I couldn't find anyone who has reported this issue, that's why it took me so long to find it. I do think I understand why it happens though, it's because if you want to build a final static binary you need to pass `--ghc-options=-optl=-static` which passes `-static` to all linker stages which in turn makes the template haskell phase fail. Some people suggested that there was a way to get template haskell to run without requiring dynamic linking but I couldn't work it out or find anyone who could work it out.